keras/layers/noise.py

```diff
@@ -20,4 +20,26 @@ def get_output(self, train=False):
 
     def get_config(self):
         return {"name":self.__class__.__name__,
-            "sigma":self.sigma}
\ No newline at end of file
+            "sigma":self.sigma}
+
+class GaussianDropout(Layer):
+    '''
+        Multiplicative Gaussian Noise
+        Reference: 
+            Dropout: A Simple Way to Prevent Neural Networks from Overfitting
+            Srivastava, Hinton, et al. 2014
+    '''
+    def __init__(self, p):
+        super(GaussianDropout,self).__init__()
+        self.p = p
+
+    def get_output(self, train):
+        X = self.get_input(train)
+        if train:
+            # self.p refers to drop probability rather than retain probability (as in paper) to match Dropout layer syntax
+            X *= srng.normal(size=X.shape, avg = 1., std=T.sqrt(self.p/(1-self.p)), dtype=theano.config.floatX)
```

Comment:
> `T.sqrt(self.p/(1-self.p))`

Something should be done to avoid division by zero in case p = 1... maybe clipping p to [epsilon, 1-epsilon]?

In terms of coding style, please use spaces around operators and use floats in float operations.